### PR TITLE
Tweak/dynamic imports webpack5 - ED-451, ED-672

### PR DESCRIPTION
**Pull request by rotemee**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: .grunt-config/webpack.js [hunk 1]
```diff
@@ -70,13 +70,16 @@ const entry = {
...
 	'app-packages': path.resolve( __dirname, '../core/app/assets/js/app-packages' ),
 	'beta-tester': path.resolve( __dirname, '../assets/dev/js/admin/beta-tester/beta-tester.js' ),
-	'frontend': path.resolve( __dirname, '../assets/dev/js/frontend/frontend.js' ),
 	'common-modules': path.resolve( __dirname, '../core/common/assets/js/modules' ),
 	'editor-modules': path.resolve( __dirname, '../assets/dev/js/editor/modules.js' ),
 	'editor-document': path.resolve( __dirname, '../assets/dev/js/editor/editor-document.js' ),
-	'frontend-modules': path.resolve( __dirname, '../assets/dev/js/frontend/modules.js' ),
 	'qunit-tests': path.resolve( __dirname, '../tests/qunit/main.js' ),
 };
+const frontendEntries = {
```
Comment (kobizz) [inline .grunt-config/webpack.js:78]: Add an extra line

File: .grunt-config/webpack.js [hunk 2]
```diff
@@ -140,12 +173,24 @@ const webpackProductionConfig = Object.assign( {}, baseConfig, {
 
 // Add minified entry points
 for ( const entryPoint in entry ) {
-	webpackProductionConfig.entry[ entryPoint ] = entry[ entryPoint ];
-	webpackProductionConfig.entry[ entryPoint + '.min' ] = entry[ entryPoint ];
+	let entryValue = entry[ entryPoint ];
+
+	if ( entryValue.dependOn ) {
+		entryValue = { ...entryValue };
```
Comment (kobizz) [inline .grunt-config/webpack.js:179]: Add a comment

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/frontend/frontend.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import '../public-path';
```
Comment (kobizz) [inline assets/dev/js/frontend/frontend.js:1]: Move it below the comment

File: includes/frontend.php [hunk 4]
```diff
@@ -447,6 +456,16 @@ public function register_scripts() {
 			true
 		);
 
+		wp_register_script(
```
Comment (kobizz) [inline includes/frontend.php:459]: Actually, it's better to register it only when enabled. You can skip the 'register' level and immediately do 'enqueue' with all the parameters.

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — DISMISSED** [review]

File: .grunt-config/webpack.js [hunk 5]
```diff
@@ -106,46 +110,115 @@ const baseConfig = {
...
+	{
+		...prodSharedConfig,
+		name: 'frontend',
+		entry: {
+			...frontendEntries,
+		},
+		optimization: {
+			...prodSharedOptimization,
+			runtimeChunk: {
+				name: 'runtime.min',
```
Comment (KingYes) [inline .grunt-config/webpack.js:189]: `webpack.runtime`

**Review by KingYes — CHANGES_REQUESTED** [review]

**Review by kobizz — DISMISSED** [review]

File: .grunt-config/remove-chunks.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+	removeChunks() {
+		if ( ! this.chunkSuffix ) {
+			return;
+		}
+
+		const chunksFolder = path.resolve( __dirname, '../assets/js' );
+
+		fs.readdir( chunksFolder, ( err, files ) => {
+			if (err) throw err;
```
Comment (kobizz) [inline .grunt-config/remove-chunks.js:21]: Add spaces

File: .grunt-config/remove-chunks.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+			for ( const fileName of files ) {
+				const filePath = path.join( chunksFolder, fileName );
+
+				if ( fileName.indexOf( this.chunkSuffix ) > -1 && fs.existsSync( filePath ) ) {
+					fs.unlink( filePath, err => {
+						if (err) throw err;
```
Comment (kobizz) [inline .grunt-config/remove-chunks.js:28]: Add spaces

**Review by kobizz — CHANGES_REQUESTED** [review]

File: .grunt-config/webpack.js [hunk 8]
```diff
@@ -101,51 +107,127 @@ const baseConfig = {
...
+
+			entryValue.dependOn += '.min';
+		}
+
+		delete config.entry[ entryPoint ];
+		config.entry[ entryPoint + '.min' ] = entryValue;
+	}
+} );
+
+const defaultConfig = webpackConfig.map( ( config ) => {
```
Comment (matipojo) [inline .grunt-config/webpack.js:224]: It should be the opposite, the default should be `watch: false` and the development should add `watch: true`.

File: assets/dev/js/frontend/elements-handler.js [hunk 9]
```diff
@@ -1,45 +1,29 @@
-import Accordion from './handlers/accordion';
-import Alert from './handlers/alert';
-import Counter from './handlers/counter';
-import Progress from './handlers/progress';
-import Tabs from './handlers/tabs';
-import Toggle from './handlers/toggle';
-import Video from './handlers/video';
-import ImageCarousel from './handlers/image-carousel';
-import TextEditor from './handlers/text-editor';
+import globalHandler from './handlers/global';
```
Comment (matipojo) [inline assets/dev/js/frontend/elements-handler.js:1]: Consider rename to `elements-async-handlers`. and `external-handlers.js` => `elements-sync-handlers`. 
or any names that explains the different between `elements-handler.js` vs `external-handlers.js`

File: includes/settings/settings.php [hunk 10]
```diff
@@ -532,6 +532,19 @@ protected function create_tabs() {
...
+							'optimized_js_loading' => [
+								'label' => __( 'Optimized JS Loading', 'elementor' ),
+								'field_args' => [
+									'type' => 'select',
+									'options' => [
+										'' => __( 'Default', 'elementor' ),
+										'enabled' => __( 'Enable', 'elementor' ),
+										'disabled' => __( 'Disable', 'elementor' ),
+									],
+									'desc' => __( 'Developers, Please Note! Optimized JS mode reduces the amount of code that is loaded on the page by default. When activated, the optimized mode allows you to dynamically load code only when it’s needed.', 'elementor' )
```
Comment (matipojo) [inline includes/settings/settings.php:544]: `Developers`?

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]

File: .grunt-config/remove-chunks.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+			fs.readdir( chunksFolder, ( err, files ) => {
+				if ( err ) throw err;
+
+				for ( const fileName of files ) {
+					const filePath = path.join( chunksFolder, fileName );
+
+					if ( fileName.indexOf( this.chunkSuffix ) > -1 && fs.existsSync( filePath ) ) {
+						fs.unlink( filePath, err => {
+							if ( err ) throw err;
+						});
```
[the remark this answers is not in the document]
Comment (rotemee) [reply .grunt-config/remove-chunks.js]: Fixed.

File: .grunt-config/remove-chunks.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+		const chunksFolder = path.resolve( __dirname, '../assets/js' );
+
+		if ( fs.existsSync( chunksFolder ) ) {
+			fs.readdir( chunksFolder, ( err, files ) => {
+				if ( err ) throw err;
+
+				for ( const fileName of files ) {
+					const filePath = path.join( chunksFolder, fileName );
+
+					if ( fileName.indexOf( this.chunkSuffix ) > -1 && fs.existsSync( filePath ) ) {
```
[the remark this answers is not in the document]
Comment (rotemee) [reply .grunt-config/remove-chunks.js]: Fixed.

File: .grunt-config/webpack.js (hunk 8 above)
Comment (rotemee) [reply .grunt-config/webpack.js:224]: The name of the process will be handled on a different PR.

File: assets/dev/js/frontend/elements-handler.js (hunk 9 above)
Comment (rotemee) [reply assets/dev/js/frontend/elements-handler.js:1]: Fixed.

File: includes/settings/settings.php (hunk 10 above)
Comment (rotemee) [reply includes/settings/settings.php]: Fixed.

**Review by matipojo — APPROVED** [review]

**Pull request merged**